invenio_records_rest/__init__.py

```diff
@@ -22,7 +22,634 @@
+  "used to group the documents by a certain criterion".
+- `metric <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations-metrics.html>`_
+  "used to create different types of metrics from values extracted from the documents
+  being aggregated".
+- `matrix <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations-matrix.html>`_
+  "used to produce a matrix result based on the values extracted from the requested document fields".
+- `pipeline <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-aggregations-pipeline.html>`_
+  "used to aggregate the output of other aggregations and their associated metrics".
+
+
+Filters
++++++++
+Filters can be applied at different points of a query having different effects
+for aggregations, but not for search queries.
+
+
+To set the aggregations and filters you want you can modify the
+`RECORDS_REST_FACETS`.
+
+>>> from .facets import terms_filter
+>>> app.config['RECORDS_REST_FACETS'] = {
+...     'index_name': {
+...         'aggs': {
+...             'type': {'terms': {'field': 'type'}}
+...         },
+...         'post_filters': {
+...             'type': terms_filter('type'),
+...         },
+...         'filters': {
+...             'typefilter': terms_filter('type'),
+...         }
+...     }
+... }
+
+Sorting
+~~~~~~~
+Sorting is based by default on a relevance score, but this can
+be configured as well. The following ways are possible:
+
+- `Sorting by field values <https://www.elastic.co/guide/en/elasticsearch/guide/current/_sorting.html#_sorting_by_field_values>`_
+- `Multilevel sorting <https://www.elastic.co/guide/en/elasticsearch/guide/current/_sorting.html#_multilevel_sorting>`_
+- `Sorting on multilevel fields <https://www.elastic.co/guide/en/elasticsearch/guide/current/_sorting.html#_sorting_on_multivalue_fields>`_
+
+This can be configured through the `RECORDS_REST_SORT_OPTIONS`:
+
+>>> RECORDS_REST_SORT_OPTIONS = dict(
+...    records=dict(
+...        bestmatch=dict(
+...            title='Best match',
+...            fields=['_score'],
+...        default_order='desc',
+...            order=1,
+...        ),
+...        mostrecent=dict(
+...        title='Most recent',
+...            fields=['_created'],
+...            default_order='asc',
+...            order=2,
+...        ),
+...    )
+... )
+
+The default configuration will return the results sorted by the best match when
+filtering by a given query, or sorted by their creation date when querying
+all results:
+
+>>> RECORDS_REST_DEFAULT_SORT = dict(
+...    records=dict(
+...        query='bestmatch',
+...        noquery='mostrecent',
+...    )
+... )
+
+Query parser
+~~~~~~~~~~~~
+
+The search syntax for the various queries is powered by the query parser used.
+Internally the query parser is referred to as the search factory.
+This defaults to the `Q() <http://elasticsearch-dsl.readthedocs.io/en/latest/search_dsl.html#queries>`_ parser
```

Comment:
I think it's better to point to the [ES Query String](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-query-string-query.html) docs here since technically this [is used if a value is passed](https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/query.py#L46-L48)